<launch>
	<!-- Common parameters -->
	<arg name="camera_name" default="camera" />
	<arg name="frame_rate" default="30" />

	<arg name="left_camera_serial" default="14150439" />
	<arg name="left_camera_calibrated" default="0" />

	<arg name="right_camera_serial" default="14150441" />
	<arg name="right_camera_calibrated" default="0" />

	<group ns="$(arg camera_name)" >

		<group ns="right" >
			<node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager"  output="screen" />

			<node pkg="nodelet" type="nodelet" name="camera_nodelet"  output="screen"
				  args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
				<param name="frame_id" value="camera_right" />
				<param name="serial" value="$(arg right_camera_serial)" />

				<!-- When unspecified, the driver will use the default framerate as given by the
                     camera itself. Use this parameter to override that value for cameras capable of
                     other framerates. -->
				<param name="frame_rate" value="$(arg frame_rate)" />

				<!-- Use the camera_calibration package to create this file -->
				<param name="camera_info_url" if="$(arg right_camera_calibrated)"
					   value="file://$(env HOME)/.ros/camera_info/$(arg right_camera_serial).yaml" />
			</node>

			<node pkg="nodelet" type="nodelet" name="image_proc_debayer" output="screen"
				  args="load image_proc/debayer camera_nodelet_manager">
			</node>

			    <node pkg="teleop_vision" type="extrinsic_calib_aruco" name="extrinsic_aruco" output="screen" >
				<rosparam command="load" file="$(find teleop_vision)/launch/params_aruco_board.yaml" />
				<param name= "cam_name" value= "flea$(arg right_camera_serial)" /> <!-- ATTENTION USING the same file for right-->
				<param name="image_transport_namespace" value="/$(arg camera_name)/right/image_color"/>
				<param name="board_to_cam_pose_topic_name" value="/$(arg camera_name)/right/board_to_camera"/>
				<param name="show_image" value="true"/>
				<param name="frequency" value="30"/>
		<!--                <param name="image_transport" value="compressed"/> --> <!-- Remove if image is not received over network -->
			    </node>

<!--			<node pkg="teleop_vision" type="extrinsic_calib_aruco" name="extrinsic_aruco" output="screen" >
				<rosparam command="load" file="$(find teleop_vision)/launch/params_aruco_board_4_4.yaml" />
				<param name="cam_intrinsic_calibration_file_path" value="$(find teleop_vision)/resources/flea_right_intrinsic.xml" />
				<param name="image_transport_namespace" value="/$(arg camera_name)/right/image_color/"/>
				<param name="board_to_cam_pose_topic_name" value="board_to_camera"/>
				<param name="draw_axes" value="true"/>
				<param name="frequency" value="30"/>
				<param name="image_transport" value="compressed"/>
			</node>
	-->		<!--<param name="/$(arg camera_name)/right/image_color/resized/downscaling_factor" value="4"/>-->

		</group>

		<group ns="left" >
			<node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" output="screen"/>

			<!-- Both cameras are not loaded into one nodelet manager to avoid the namespacing issue. -->
			<node pkg="nodelet" type="nodelet" name="camera_nodelet"  output="screen"
				  args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
				<param name="frame_id" value="camera_left" />
				<param name="serial" value="$(arg left_camera_serial)" />

				<!-- When unspecified, the driver will use the default framerate as given by the
                     camera itself. Use this parameter to override that value for cameras capable of
                     other framerates. -->
				<param name="frame_rate" value="$(arg frame_rate)" />

				<!-- Use the camera_calibration package to create this file -->
				<param name="camera_info_url" if="$(arg left_camera_calibrated)"
					   value="file://$(env HOME)/.ros/camera_info/$(arg left_camera_serial).yaml" />
			</node>

			<node pkg="nodelet" type="nodelet" name="image_proc_debayer"  output="screen"
				  args="load image_proc/debayer camera_nodelet_manager">
			</node>



			    <node pkg="teleop_vision" type="extrinsic_calib_aruco" name="extrinsic_aruco" output="screen" >
				<rosparam command="load" file="$(find teleop_vision)/launch/params_aruco_board.yaml" />
				<param name= "cam_name" value= "flea$(arg left_camera_serial)" /> <!-- ATTENTION USING the same file for right-->
				<param name="image_transport_namespace" value="/$(arg camera_name)/left/image_color"/>
				<param name="board_to_cam_pose_topic_name" value="/$(arg camera_name)/left/board_to_camera"/>
				<param name="show_image" value="true"/>
				<param name="frequency" value="30"/>
		<!--                <param name="image_transport" value="compressed"/> --> <!-- Remove if image is not received over network -->
			    </node>

		</group>



		<node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc">
			<param name="approximate_sync" value="true"/>
		</node>

	</group>
</launch>
